pkg/controllers/consistency/controller.go

```diff
@@ -78,39 +79,47 @@ func (c *Controller) Name() string {
 	return "consistency"
 }
 
-func (c *Controller) Reconcile(ctx context.Context, node *v1.Node) (reconcile.Result, error) {
-	if node.Labels[v1alpha5.ProvisionerNameLabelKey] == "" {
+func (c *Controller) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
+	if machine.Status.ProviderID == "" {
 		return reconcile.Result{}, nil
 	}
-	// If we get an event before we should check for consistency checks, we ignore and wait
-	if lastTime, ok := c.lastScanned.Get(client.ObjectKeyFromObject(node).String()); ok {
+	// If we get an event before we should check for inflight checks, we ignore and wait
+	if lastTime, ok := c.lastScanned.Get(client.ObjectKeyFromObject(machine).String()); ok {
 		if lastTime, ok := lastTime.(time.Time); ok {
 			remaining := scanPeriod - c.clock.Since(lastTime)
 			return reconcile.Result{RequeueAfter: remaining}, nil
 		}
 		// the above should always succeed
 		return reconcile.Result{RequeueAfter: scanPeriod}, nil
 	}
-	c.lastScanned.SetDefault(client.ObjectKeyFromObject(node).String(), c.clock.Now())
+	c.lastScanned.SetDefault(client.ObjectKeyFromObject(machine).String(), c.clock.Now())
 
+	node, err := machineutil.NodeForMachine(ctx, c.kubeClient, machine)
```

Comment:
Can you add a comment here describing that we don't care about the DuplicateNode or NodeNotFound because we reconcile this with another controller?

Reply:
Sure, in general this should be an invariant and, if it's violated, should be handled by the statusConditions of the machine lifecycle controller